handler.go

```diff
@@ -126,19 +126,19 @@ func (t *myTransport) RoundTrip(request *http.Request) (*http.Response, error) {
 	for _, parsedRequest := range parsedRequests {
 		if !t.AllowLimit(parsedRequest) {
 			log.Println("User hit the limit:", parsedRequest.Path, " from IP: ", parsedRequest.RemoteAddr)
-			return jsonRPCError(parsedRequest.ID, -32000, "You hit the request limit")
+			return jsonRPCError(parsedRequest.ID, -32000, http.StatusTooManyRequests, "You hit the request limit")
 		}
 
 		if !t.MatchAnyRule(parsedRequest) {
 			log.Println("Not allowed:", parsedRequest.Path, " from IP: ", parsedRequest.RemoteAddr)
-			return jsonRPCError(parsedRequest.ID, -32601, "You are not authorized to make this request")
+			return jsonRPCError(parsedRequest.ID, -32601, http.StatusUnauthorized, "You are not authorized to make this request")
 		}
 	}
 	request.Host = request.RemoteAddr //workaround for CloudFlare
 	response, err = http.DefaultTransport.RoundTrip(request)
 	if err != nil {
 		log.Println("Error response from RoundTrip:", err)
-		return jsonRPCError(parsedRequests[0].ID, -32603, "Internal error") //returning ID of the first request
+		return jsonRPCError(parsedRequests[0].ID, -32603, response.StatusCode, "Internal error") //returning ID of the first request
```

Comment:
Does this status code (or even response) always necessarily exist? Or could `RoundTrip` return only an `err`?